src/sql/workbench/parts/objectExplorer/browser/iconRenderer.ts

```diff
@@ -0,0 +1,371 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { generateUuid } from 'vs/base/common/uuid';
+
+export function renderServerIcon(element: HTMLElement, serverInfo: ServerInfo, isConnected: boolean,
+	iconSizePx?: number, topMarginPx?: number, leftMarginPx?: number
+): void {
+	let serverIcon: IconInfo = ServerIcon.getServerIcon(serverInfo);
+	renderServerIconByPath(element, serverIcon, isConnected, iconSizePx, topMarginPx, leftMarginPx);
+}
+
+export function renderServerIconByPath(element: HTMLElement, iconData: IconPath | IconInfo, isConnected: boolean,
+	iconSizePx?: number, topMarginPx?: number, leftMarginPx?: number
+): void {
+	if (!element) { return; }
+
+	let iconPath: IconPath = undefined;
+	if (iconData['path']) {
+		let iconInfo = iconData as IconInfo;
+		iconPath = iconInfo.path;
+		iconSizePx = iconSizePx || Math.min(iconInfo.widthPx, iconInfo.heightPx);
```

Comment:
Let's just make them always 16x16 if possible. Seems like VSCode always does this?

Reply:
fixed.